components/kyma-environment-broker/internal/model.go

```diff
@@ -124,11 +128,11 @@ type ProvisioningOperation struct {
 type DeprovisioningOperation struct {
 	Operation `json:"-"`
 
-	ProvisioningParameters string           `json:"provisioning_parameters"`
-	Avs                    AvsLifecycleData `json:"avs"`
-	EventHub               EventHub         `json:"eh"`
-	SubAccountID           string           `json:"-"`
-	RuntimeID              string           `json:"runtime_id"`
+	ProvisioningParameters string             `json:"provisioning_parameters"`
+	Avs                    AvsLifecycleData   `json:"avs"`
+	Azure                  AzureLifecycleData `json:"azure"`
```

Comment:
I discussed this with @piotrmiskiewicz and the conclusion we came to was that in this case migration won't be necessary. During the deprovisioning of SKRs created before this change, there is no information stored in `eh` field that would be required for deprovisioning (as it's using only the label to identify resource group it needs to delete, and that didn't change).

Reply:
I'm thinking about it. Is it possible that a deprovisioning is started with old KEB, then upgrade KEB, then deprovisioning is rescheduled. That is the only risk.

Reply:
In such case, the worst thing that could happen would be:
1. Old KEB removes Resource Group, sets the `deleted` operation flag to `true`
2. KEB gets upgraded
3. New KEB checks if `deleted` flag is set to true, it's not (because the json changed)
4. New KEB tries deleting the resources again: the step starts with checking if resource group with specific label exists, it doesn't because it was deleted by "old KEB", step is finished again as success
5. Deprovisioning continues without any issues

Right?

Reply:
right, for me it is ok as it is now :)

Reply:
You are right, nothing should break. Unmarshalling will work and set the default values for the AzureLifecycleData struct.